### ISSUE TITLE
Yaw, Roll, and Pitch

### ISSUE THREAD

**Issue by Stroker63** [sheepsskullcity/opentrackkinect#7]
Hello, i got the thing working, but i need to get the yaw, roll and pitch to work in order to fully play.
Do you have any idea on how i could get them working?
Im also using my cellphone for the yaw/roll/pitch, but it doesnt work at the same time with the kinect, since they both use the same protocol in opentrack.
Thanks a lot of this! Do you know if you coul help me?

**Comment by ajayyy** [comment]
Are you using Vridge?

**Comment by Stroker63** [comment]
Nope, I'm using an app called HeadTracker, basically send the phone's sensors data to opentrack

**Comment by ajayyy** [comment]
Oh ok, what are you using it with? Flight sim or vr?

**Comment by Stroker63** [comment]
Flight Sims, especially DCS and War Thunder

**Comment by ajayyy** [comment]
Oh, I see. I think it was designed for vr with Vridge, that already has rotation I'll take a quick look into the code to see how easy it would be to add support

**Comment by ajayyy** [comment]
Wait, can't you have multiple sensors in opentrack?

**Comment by Stroker63** [comment]
Thanks a lot! I also found an alternative, but I'm unsure if it would work at all.
So, FaceTrackNoIR has support for 2 sources of tracking, but the application sends data to the port 4242, and facetrack doesn't go anything lower than 5550.
I also tried using an older version of opentrack that allows 2 sources, but the actual tracking was jumping all around, like if it was confused of what tracker to use

**Comment by ajayyy** [comment]
Oh I see

**Comment by ajayyy** [comment]
Changing the port seems like it's a one line change, so you could try that. https://github.com/sheepsskullcity/opentrackkinect/blob/master/src/org/opentrackkinect/Main.java#L70

Do you know how to compile Java programs? If not, I can do it, but not until tomorrow.

**Comment by Stroker63** [comment]
I tried that, there are 2 lines with the port tho.
But to be honest I didn't know anything about java programing until like a week ago, and it didn't matter since I couldn't get it to work.
I would really thank you if you could do it, and yes there is no problem in waiting

**Comment by ajayyy** [comment]
Ok, I can try. I haven't actually used this program in a while, and I didn't make it. But, it gave me a notification, so I thought I'd help you out, so sorry if it doesn't work. If I don't do it tomorrow, you can remind me, I might've forgotten.

**Comment by Stroker63** [comment]
Don't worry man! And thanks for giving me some of your time!

**Comment by ajayyy** [comment]
Sorry, don't have time tonight, will do it tomorrow.

**Comment by ajayyy** [comment]
Sorry I completely forgot!! Now, I forget what I was trying to do. Was I trying to change the port to something for FreeTrackNoIR? And will that allow 2 sources?

**Comment by Stroker63** [comment]
Crap I completely forgot too!
Yeah, the problem is that the app uses the 4242 port, but facetrack can
only lower it's port down to 5000 or something.
Changing the port from 4242 to 5555 should do the trick just fine.
Thanks for reminding me!

**Comment by ajayyy** [comment]
@Stroker63 I know I'm probably **WAAAAYYY** too late, sorry about that. But here is a recompiled version with the port variable changed. I could not find the second line you were talking about though. When doing CTRL+F, it only references 4242 once.

https://github.com/ajayyy/opentrackkinect/releases/tag/v1.0.2

Sorry for it being so late. Hopefully it still helps!

If you need anything else, just ask. I'll be sure to be much faster next time :).

**Comment by Stroker63** [comment]
Man thanks! And dont worry, glad you got back to it.
Now, what you did worked nicely, but seems like facetrack doesnt like 2 tracking sources in the same port, so it doesnt work. Guess ill have to wait for an application that lets me combine multiple tracking sources.
Thanks again man! And as for the second 4242 i have no idea, i guess i was doing something wrong haha.

**Comment by ajayyy** [comment]
Do you want it at port 5556 so it's a different port?

**Comment by Stroker63** [comment]
Oh no, the port where is at works just fine, seems like facetrack doesnt allow for multi-port.
But opentrack got updated like a week ago and you can fusion 2 traacking sources and i just realized, just that i cant use the UDP over network twice, a workaround should be using freepie on my phone, but i cant get it to work. No idea what to o next, but at least there is hope now

**Comment by ajayyy** [comment]
That's good. Sucks that it won't work though. Good thing there's a chance.

**Comment by erpadmin** [comment]
should yaw pitch roll actually work natively though?  i ran the sdk 2 sample face basic and it shows that it registers these values
